.github/workflows/test-linea-devnet.yml

```diff
@@ -0,0 +1,134 @@
+name: test-linea-devnet
+
+on:
+  pull_request:
+    branches:
+      - main
+
+  workflow_call:
+    outputs:
+      workflow_run_id: 
+        description: "the run id of the workflow"
+        value: ${{ jobs.build.outputs.workflow_run_id }}
+  workflow_dispatch:
+
+permissions:
+  id-token: write  
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: dockerhub
+    outputs:
+      workflow_run_id: ${{ steps.assemble.outputs.workflow_id }}
+      dockertag: ${{ steps.assemble.outputs.dockertag }}
+      dockerimage: ${{ steps.assemble.outputs.dockerimage }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: build the linea devnet artifacts
+        id: assemble
+        uses: ./.github/actions/assemble-devnet
+
+      - name: set up docker buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker        
+
+      - name: login to docker
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USER_RW }}
+          password: ${{ secrets.DOCKER_PASSWORD_RW }}
+
+      - name: set docker build args
+        run: |
+          echo "Building docker tag: ${{ steps.assemble.outputs.dockertag }}"
+          echo "Building docker image: ${{ steps.assemble.outputs.dockerimage }}"
+
+      - name: build the image
+        uses: docker/build-push-action@v6
+        env:
+          DOCKER_BUILD_SUMMARY: false
+        with:
+          context: linea-besu/.
+          platforms: linux/amd64
+          provenance: false
+          build-args: |
+            VERSION=${{ steps.assemble.outputs.dockertag }}
+            VCS_REF=${{ github.sha }}
+            BUILD_DATE=${{ steps.assemble.outputs.build_date }}
+          tags: |
+            consensys/linea-besu-package:${{ steps.assemble.outputs.dockertag }}
+
+      - name: Save Docker image
+        id: dockersave
+        run: |
+          docker images
+          docker save consensys/linea-besu-package:${{ steps.assemble.outputs.dockertag }} -o /tmp/docker-image-devnet.tar
+
+      - name: Upload artifact
```

Comment:
Why do we need to upload it though? Can't we just continue on with the testing in the same job? We're not reusing steps across workloads anyway